modules/service/internal/types/schema.go

```diff
@@ -0,0 +1,184 @@
+		return err
+	}
+
+	return nil
+}
+
+// ValidateBindingPricing checks if the given pricing is valid
+func ValidateBindingPricing(pricing string) error {
+	// TODO
+	return nil
+}
+
+// ValidateRequestInput validates the request input against the input schema.
+// ensure that the schemas is valid and input is a valid JSON
+func ValidateRequestInput(schemas string, input string) error {
+	inputSchemaBz := parseInputSchema(schemas)
+
+	if !validateDocument(inputSchemaBz, input) {
+		return sdkerrors.Wrap(ErrInvalidRequestInput, "invalid request input")
+	}
+
+	return nil
+}
+
+// ValidateResponseOutput validates the response output against the output schema.
+// ensure that the schemas is valid and output is a valid JSON
+func ValidateResponseOutput(schemas string, output string) error {
+	outputSchemaBz := parseOutputSchema(schemas)
+
+	if !validateDocument(outputSchemaBz, output) {
+		return sdkerrors.Wrap(ErrInvalidResponseOutput, "invalid response output")
+	}
+
+	return nil
+}
+
+// ValidateResponseError validates the response err against the error schema.
+// ensure that the schemas is valid and errResp is a valid JSON
+func ValidateResponseError(schemas string, errResp string) error {
+	errSchemaBz := parseErrorSchema(schemas)
+
+	if !validateDocument(errSchemaBz, errResp) {
+		return sdkerrors.Wrap(ErrInvalidResponseErr, "invalid response err")
+	}
+
+	return nil
+}
+
+func validateInputSchema(inputSchema map[string]interface{}) error {
+	inputSchemaBz, err := json.Marshal(inputSchema)
+	if err != nil {
+		return sdkerrors.Wrapf(ErrInvalidSchemas, "invalid input schema: %s", err)
+	}
+
+	_, err = gojsonschema.NewSchema(gojsonschema.NewBytesLoader(inputSchemaBz))
+	if err != nil {
+		return sdkerrors.Wrapf(ErrInvalidSchemas, "invalid input schema: %s", err)
+	}
+
+	return nil
+}
+
+func validateOutputSchema(outputSchema map[string]interface{}) error {
+	outputSchemaBz, err := json.Marshal(outputSchema)
+	if err != nil {
+		return sdkerrors.Wrapf(ErrInvalidSchemas, "invalid output schema: %s", err)
+	}
+
+	_, err = gojsonschema.NewSchema(gojsonschema.NewBytesLoader(outputSchemaBz))
+	if err != nil {
+		return sdkerrors.Wrapf(ErrInvalidSchemas, "invalid output schema: %s", err)
+	}
+
+	return nil
+}
+
+func validateErrorSchema(errSchema map[string]interface{}) error {
+	errSchemaBz, err := json.Marshal(errSchema)
+	if err != nil {
+		return sdkerrors.Wrapf(ErrInvalidSchemas, "invalid error schema: %s", err)
+	}
+
+	_, err = gojsonschema.NewSchema(gojsonschema.NewBytesLoader(errSchemaBz))
+	if err != nil {
+		return sdkerrors.Wrapf(ErrInvalidSchemas, "invalid error schema: %s", err)
+	}
+
+	return nil
+}
+
+// parseInputSchema parses the input schema from the given schemas.
+// ensure that the schemas is valid. Panic if invalid
+func parseInputSchema(schemas string) []byte {
+	var svcSchemas ServiceSchemas
+	if err := json.Unmarshal([]byte(schemas), &svcSchemas); err != nil {
+		panic(err)
```

Comment:
why panic?

Reply:
My consideration is that the param will be retrieved from store when the method is called from keeper (the stored schemas must be valid).  No panicking is better for common use (e.g. client).

Reply:
But it is also used in the client.